src/SoCreate.Extensions.Logging/LogAdapters/SqlServerLoggerLogConfigurationAdapter.cs

```diff
@@ -23,13 +23,18 @@ public SqlServerLoggerLogConfigurationAdapter(IConfiguration configuration)
 
         public LoggerConfiguration ApplyConfiguration(LoggerConfiguration loggerConfiguration, ActivityLoggerOptions activityLoggerOptions)
         {
-            var sqlConnectionString = activityLoggerOptions.SqlServer?.ConnectionString ??
-                                      _configuration.GetValue<string>("Infrastructure:ConnectionString");
-            var tableName = activityLoggerOptions.SqlServer?.TableName ?? "Activity";
-            var schemaName = activityLoggerOptions.SqlServer?.SchemaName ?? "Logging";
+            var sqlConnectionString = string.IsNullOrEmpty(activityLoggerOptions.SqlServer?.ConnectionString)
+                ? _configuration.GetValue<string>("Infrastructure:ConnectionString")
+                : activityLoggerOptions.SqlServer?.ConnectionString;
+            var tableName = string.IsNullOrEmpty(activityLoggerOptions.SqlServer?.TableName)
+                ? "Activity"
+                : activityLoggerOptions.SqlServer?.TableName;
+            var schemaName = string.IsNullOrEmpty(activityLoggerOptions.SqlServer?.SchemaName)
+                ? "Logging"
+                : activityLoggerOptions.SqlServer?.SchemaName;
             var logType = activityLoggerOptions.ActivityLogType;
 
-            if (sqlConnectionString == null)
+            if (sqlConnectionString == string.Empty)
```

Comment:
Is this always going to be a string?

Reply:
Yes, thats how I found this problem, when you call `_configuration.GetValue<string>` it returns a string, so there is no way for it to be null